src/components/ContentHighlights/CurrentContentHighlightHeader.jsx

```diff
@@ -1,28 +1,76 @@
-import React from 'react';
+import React, { useState, useEffect, useContext } from 'react';
 import {
-  Button, ActionRow,
+  Button, ActionRow, Alert,
 } from '@edx/paragon';
-import { Add } from '@edx/paragon/icons';
-
+import { Add, Info } from '@edx/paragon/icons';
+import { EnterpriseAppContext } from '../EnterpriseApp/EnterpriseAppContextProvider';
 import { useContentHighlightsContext } from './data/hooks';
-import { BUTTON_TEXT, HEADER_TEXT } from './data/constants';
+import {
+  BUTTON_TEXT, HEADER_TEXT, MAX_HIGHLIGHT_SETS_PER_ENTERPRISE_CURATION, ALERT_TEXT,
+} from './data/constants';
 
 const CurrentContentHighlightHeader = () => {
+  const {
+    enterpriseCuration: {
+      enterpriseCuration: {
+        highlightSets,
+      },
+    },
+  } = useContext(EnterpriseAppContext);
   const { openStepperModal } = useContentHighlightsContext();
+  // Preliminiary logic for the header text given max sets reached
+  const [maxHighlightsReached, setMaxHighlightsReached] = useState(false);
+  const [showAlert, setShowAlert] = useState(false);
+
+  const createNewHighlight = () => {
+    if (maxHighlightsReached) {
+      setShowAlert(true);
+    } else {
+      openStepperModal();
+    }
+  };
+  useEffect(() => {
+    // using greater than or equal as an additional buffer as opposed to exactly equal
+    if (highlightSets.length >= MAX_HIGHLIGHT_SETS_PER_ENTERPRISE_CURATION) {
+      setMaxHighlightsReached(true);
+    } else {
+      setMaxHighlightsReached(false);
+    }
+  }, [highlightSets, setMaxHighlightsReached]);
```

Comment:
nit: was linting warning you about putting `setMaxHighlightsReached` in the dependency array of this `useEffect`? Typically, the state setter function from `useState` doesn't have to be included since its memoized by React already.